clash-prelude/src/Clash/Prelude/DataFlow.hs

```diff
@@ -18,7 +18,7 @@ Self-synchronizing circuits based on data-flow principles.
 {-# OPTIONS_GHC -fplugin GHC.TypeLits.Normalise       #-}
 {-# OPTIONS_HADDOCK show-extensions #-}
 
-module Clash.Prelude.DataFlow
+module Clash.Prelude.DataFlow {-# DEPRECATED "Module will be removed in future versions of clash-prelude." #-}
```

Comment:
May or may not be the time for this, but I feel like we should commit to some fixed offset between deprecating something and removing it. The reason being

  * users would know how quickly they need to change their code before it breaks
  * we could better ensure deprecated code isn't removed too quickly / slowly

Thoughts?

Reply:
Agreed. Ideally we should point users to some alternative.

Reply:
Guess I'll work a bit harder on `martijn-circuits` this weekend :-)

Reply:
Well now that you know a circuit is a _symmetric monoidal category_, what's stopping you? :wink: